### ISSUE TITLE
Implement Github actions for CI

### ISSUE THREAD

**Issue by ViralBShah** [JuliaMath/openlibm#226]
Would be nice to transition to github actions and test on mac, linux and windows. Help appreciated!

**Comment by ViralBShah** [comment]
Implemented in https://github.com/JuliaMath/openlibm/commit/9dd3049e931cbdce1825fb4bb0b3d3aa832b9786 but needs to be fixed for Windows.

**Comment by ViralBShah** [comment]
Attempting in #240 but could clearly benefit from someone who knows this stuff better than me!

**Comment by ViralBShah** [comment]
@dilumaluthge @giordano Could use your help here to get windows 32-bit testing correctly (currently disabled) - and generally taking a look at the ci file to see if I have done it right.

**Comment by DilumAluthge** [comment]
Just FYI, `@giordano` is on vacation.

Can you summarize the issue you're having with Windows 32-bit?

**Comment by DilumAluthge** [comment]
https://github.com/JuliaMath/openlibm/blob/6ea5fa240818a4ed4441b38eb0143274488f9cca/.github/workflows/ci.yml#L38-L44

I'm guessing this is the part that's causing problems?

**Comment by DilumAluthge** [comment]
Apart from that part, the CI file looks pretty normal to me.

**Comment by DilumAluthge** [comment]
Hmmm. On Windows x86, should we do `make ARCH=i386` instead of `make`?

**Comment by DilumAluthge** [comment]
So, just as an example, you might replace this:
```yaml
- name: Build and run tests
  run: make && make test
```

With this:
```yaml
- name: Build
  run: make
  if: ${{ matrix.os != 'windows-latest' || matrix.arch != 'x86' }}

- name: Build (Windows x86)
  run: make ARCH=i386
  if: ${{ matrix.os == 'windows-latest' && matrix.arch == 'x86' }}

- name: Run tests
  run: make test
```

**Comment by ViralBShah** [comment]
I approved the workflow.

**Comment by ViralBShah** [comment]
#243 closes this
